src/app/infobox/infobox.component.ts

```diff
@@ -0,0 +1,36 @@
+import {Component, OnInit, ChangeDetectorRef} from '@angular/core';
+import {Router, ActivatedRoute} from '@angular/router';
+import {Store} from '@ngrx/store';
+import * as fromRoot from '../reducers';
+import {KnowledgeapiService} from '../services/knowledgeapi.service';
+
+@Component({
+  selector: 'app-infobox',
+  templateUrl: './infobox.component.html',
+  styleUrls: ['./infobox.component.css']
+})
+export class InfoboxComponent implements OnInit {
+  public title: string;
+  public description: string;
+  query$: any;
+  resultsearch = '/search';
+  constructor(private knowledgeservice: KnowledgeapiService, private route: Router, private activatedroute: ActivatedRoute,
+              private store: Store<fromRoot.State>, private ref: ChangeDetectorRef) {
```

Comment:
Follow clean code quality as in Susper and Loklak, and write each property in separate lines in constructor, it makes easier for new comers and other developers to easily see what is going on in the codebase.

e.g. 
```
constructor(
    private one: FirstProperty,
    private two: SecondProperty,
    ...
) { }
```